src/components/SessionInfoBar.tsx

```diff
@@ -10,6 +16,33 @@ export function SessionInfoBar() {
     store => store.session.serverInfo
   );
   const dispatch = useAppDispatch();
+  const [isHealthy, setIsHealthy] = useState(true);
+  const client = getClient();
+
+  const checkHeartbeat = async () => {
```

Comment:
I can see the argument for why this deserves to be local state, but for the sake of consistency, I feel like this should be done through sagas. We only use the client directly like this in 2 components, compared to the many more instances where we use it in sagas.

Reply:
Ah, I've been slowly pushing towards hooks and away from sagas/redux. But I didn't make this a shared hook since it's only used in one component. We should have a bigger conversation around that. I'm a big proponent of removing redux to reduce dependencies and reduce dependency injection.

Reply:
> I've been slowly pushing towards hooks and away from sagas/redux

It's worth having a conversation about the way we (ab)use redux + sagas, but until we make a larger decision about how we want to change the way we do state + data fetching, I think it'd be good to keep things internally consistent

Reply:
Made a first attempt